shenyu-sdk/shenyu-sdk-httpclient/src/main/java/org/apache/shenyu/sdk/httpclient/HttpShenyuSdkClient.java

```diff
@@ -136,11 +144,31 @@ protected ShenyuResponse doRequest(final ShenyuRequest request) throws IOExcepti
             }
         }
 
-        HttpResponse response = getHttpClient().execute(requestBuilder.build());
-        return new ShenyuResponse(response.getStatusLine().getStatusCode(), response.getStatusLine().getReasonPhrase(),
-                Arrays.stream(response.getAllHeaders()).collect(Collectors.groupingBy(Header::getName, HashMap::new,
-                        Collectors.mapping(Header::getValue, Collectors.toCollection(LinkedList::new)))),
-                EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8), request);
+        Future<HttpResponse> execute = getHttpClient().execute(requestBuilder.build(), new FutureCallback<HttpResponse>() {
+            @Override
+            public void completed(final HttpResponse response) {
+                LOG.debug("HttpResponse completed statusLine={}", response.getStatusLine());
+            }
+
+            @Override
+            public void failed(final Exception ex) {
+                LOG.error("HttpResponse failed", ex);
+            }
+
+            @Override
+            public void cancelled() {
+                LOG.debug("HttpResponse cancelled.");
+            }
+        });
+        try {
+            HttpResponse response = execute.get();
```

Comment:
Isn't `execute.get()` a synchronous method?

Reply:
yes  it sync ,but AbstractShenyuSdkClient#execute  return ShenyuResponse .not Future<HttpResponse> so I need to get() response

Reply:
so, what's the difference with them, `httpasyncclient` and `httpclient`.

Reply:
I think this diff like reids client Lettuce sync vs Jedis.  in Thread IO model